src/ios/CDVLocation.m

```diff
@@ -80,87 +81,111 @@ - (BOOL)isAuthorized
     return YES;
 }
 
-- (BOOL)isLocationServicesEnabled
-{
-    BOOL locationServicesEnabledInstancePropertyAvailable = [self.locationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 3.x
-    BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
-
-    if (locationServicesEnabledClassPropertyAvailable) { // iOS 4.x
-        return [CLLocationManager locationServicesEnabled];
-    } else {
-        return NO;
-    }
+- (void)isLocationServicesEnabledWithCompletion:(void (^)(BOOL enabled))completion {
+    dispatch_async(queue, ^{
+        BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
+        
+        BOOL result = NO;
+        if (locationServicesEnabledClassPropertyAvailable) { // iOS 4.x
+            result = [CLLocationManager locationServicesEnabled];
+        }
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (completion) {
```

Comment:
Is this required?

Reply:
the if statement?

Reply:
yes

Reply:
That's why my comment was a question and not a sentence 😅. Nevertheless, as a rule of thumb, we should be sure before making changes.

Reply:
Well, we are not using any reference strong or weak inside the closure, so there's not really anything to make it stay alive.